lxd/instance/drivers/driver_qemu.go

```diff
@@ -5761,6 +5748,16 @@ func (d *qemu) Info() instance.Info {
 }
 
 func (d *qemu) Metrics() (*metrics.MetricSet, error) {
+	val := d.expandedConfig["security.agent.metrics"]
+
+	if val == "" || shared.IsTrue(val) {
+		return d.getAgentMetrics()
```

Comment:
@stgraber @monstermunchkin not strictly part of the PR, but a question, should we fall back to agentless metrics if the agent isn't reachable (like we do not network state above)?

Reply:
I wouldn't do this by default since some QEMU stats differ from the ones provided by the agent. We could however make this configurable.